test/compilable/extra-files/test8707.d

```diff
@@ -0,0 +1,9 @@
+struct Foo
```

Comment:
This test doesn't do a lot now, could be
```
module test;

struct Foo
{
    int i = void;
}
static assert(__traits(isZeroInit, Foo));

void main()
{
    Foo f;
    version(Win32) 
        assert(typeid(Foo).initializer.ptr >= cast(void*)&_edata && 
               typeid(Foo).initializer.ptr < cast(void*)&_edata);
}

version(Win32)
{
    extern(C) extern __gshared
    {
        int _edata;     // &_edata is start of BSS segment
        int _end;       // &_end is past end of BSS
    }
}
```

Reply:
Scratch that, a zero-initializer is not even placed into BSS! This would be a runtime check:
```
module test;

struct Foo
{
    int i = void;
}
static assert(__traits(isZeroInit, Foo));

void main()
{
    Foo f = Foo.init;
    assert(typeid(Foo).initializer.ptr == null);
}
```

Reply:
> a zero-initializer is not even placed into BSS!

Wrong again ;-/ `Foo f;` uses "memset(0)", while `Foo f = Foo.init;` copies the actual init from DATA/BSS if the struct is larger than 32 bytes (even for optimized builds).

It is not allowed to get the address of `Foo.init`. Can't a zero-initialized `init` be removed completely from DATA/BSS?